dotnet/src/SemanticKernel.Abstractions/AI/TextToImage/ImageContent.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.SemanticKernel.TextToImage;
+
+/// <summary>
+/// Represents image content.
+/// </summary>
+public class ImageContent : ContentBase
+{
+    /// <summary>
+    /// The URI of image.
+    /// </summary>
+    public Uri Uri { get; set; }
```

Comment:
Sooner or later, we will need to support images represented as byte arrays. It would be good to consider whether this will be modeled either through a new `byte[] Bytes {get; set;}` property of the class or as an `ImageByteContent` subclass of `ContentBase`. This exercise will be useful in shaping the SK public API surface now to prevent potential breaking changes in the future.

Reply:
## Points to consider 

How are we going to distinct referenced contents from non referenced ones?

### Following the abstraction path: 

`StreamingContentBase` has a `abstract byte[] ToByteArray()` method.

We might consider bringing this to a `ByteContent` abstract class or directly to the `ContentBase` to be similar.

### How to handle Referenced/Link/Anchor contents?

Any content could be referenced by a Uri, might worth having a `ILinkContent` that ImageContent/AudioContent, etc could inherit or implement

```csharp
interface ILinkContent {
    Uri Uri { get }
}

public OpenAIReferencedImageContent : ContentBase, IReferencedContent {
    Uri Uri { get; set; }
}
```

Reply:
> Sooner or later, we will need to support images represented as byte arrays. It would be good to consider whether this will be modeled either through a new `byte[] Bytes {get; set;}` property of the class or as an `ImageByteContent` subclass of `ContentBase`. This exercise will be useful in shaping the SK public API surface now to prevent potential breaking changes in the future.

Both options (new property or new separate class) look like addition, so I'm not sure there will be breaking changes when we want to extend it. In case of new property, we will have to convert `Uri` property to nullable, because it will be either image link or image byte array. 

> Any content could be referenced by a Uri, might worth having a `ILinkContent` that ImageContent/AudioContent, etc could inherit or implement

Hmm, that's interesting idea. At the moment, I can't find a scenario when `ILinkContent` abstraction will be useful for us or end-users, could you share any examples?

Reply:
> In case of new property, we will have to convert Uri property to nullable.

Making a property nullable can be considered a breaking change, right? Particularly if the existing consumers are not expecting or handling a null value for the property. Should we make it nullable now to be prepared?

Reply:
@SergeyMenshykh Yes, that makes sense. Converted `Uri` parameter to nullable.